tests/system/test_signal_crashes.py

```diff
@@ -296,9 +305,31 @@ def create_test_process(self, command=None, uid=None, args=None):
             else:
                 break
 
-        time.sleep(0.3)  # needs some more setup time
+        # sleep command needs extra time to get into "sleeping" state
+        if command == self.TEST_EXECUTABLE:
+            wait_for_sleep_cmd(process.pid)
+
         return process
 
+    def test_create_test_sleep_process(self) -> None:
+        """Test create_test_process() helper method with sleep process."""
+        proc: subprocess.Popen = self.create_test_process()
```

Comment:
Explicit type hint not needed here.
```suggestion
        proc = self.create_test_process()
```